fairroot/eventdisplay/FairEventManager.cxx

```diff
@@ -174,6 +174,11 @@ FairEventManager::~FairEventManager()
 
 void FairEventManager::Open() {}
 
+void FairEventManager::SetXMLConfig(TString xml_config)
+{
+    if (xml_config.Length() && xml_config.EndsWith(".xml"))
+        fXMLFile.reset(new FairXMLFile(xml_config));
```

Comment:
Please avoid `new` when possible.

and why force a `.xml` ending? This will be confusing to users, when they call `SetXMLConfig` and just nothing happens, just because they call their files `.XML`?

```suggestion
    if (xml_config.Length()) {
        fXMLFile = std::make_unique<FairXMLFile>(xml_config);
    }
```